app/src/main/java/ch/epfl/qedit/model/MatrixFormat.java

```diff
@@ -31,6 +35,36 @@ public MatrixFormat(
         this.hintString = hint();
     }
 
+    public static MatrixFormat parse(String format) {
+        /** Match format: 'matrixNxM' where N and M are [0-9]+ */
+        if (Pattern.compile("^(\\s*)matrix(\\d+)x(\\d+)(\\s*)$").matcher(format).find()) {
```

Comment:
You could have trimmed the string that would have been easier in my opinion

Reply:
Yes but parsing the numbers has to be done using a regexp I think